src/main/java/com/diamondfire/helpbot/bot/events/commands/SlashCommandEvent.java

```diff
@@ -0,0 +1,99 @@
+package com.diamondfire.helpbot.bot.events.commands;
+
+import com.diamondfire.helpbot.bot.command.argument.impl.parsing.exceptions.ArgumentException;
+import com.diamondfire.helpbot.bot.command.impl.Command;
+import com.diamondfire.helpbot.bot.command.reply.*;
+import com.diamondfire.helpbot.bot.command.reply.handler.*;
+import net.dv8tion.jda.api.*;
+import net.dv8tion.jda.api.entities.*;
+
+import java.util.Map;
+
+public class SlashCommandEvent implements CommandEvent {
+    private final net.dv8tion.jda.api.events.interaction.SlashCommandEvent internalEvent;
+    private final SlashReplyHandler slashReplyHandler;
+    private Map<String, ?> argumentMap;
+    private Command command;
+    
+    public SlashCommandEvent(net.dv8tion.jda.api.events.interaction.SlashCommandEvent internalEvent) {
+        this.internalEvent = internalEvent;
+        this.slashReplyHandler = new SlashReplyHandler(internalEvent);
+    }
+    
+    public void putArguments(Map<String, ?> input) {
+        argumentMap = input;
+    }
+    
+    @Override
+    public void pushArguments(String[] rawArgs) throws ArgumentException {}
```

Comment:
This is what I mean by abstracting arguments perhaps, as then you have dead code like this that doesn't apply to your system.